### PR TITLE
plugin: Add thread resolver at startup

### PR DESCRIPTION
**Pull request by Aursen**

### CODE REVIEW

File: plugin/src/plugin.rs [hunk 1]
```diff
@@ -47,6 +51,14 @@ impl GeyserPlugin for SablierPlugin {
         info!("Loading snapshot...");
         let config = PluginConfig::read_from(config_file)?;
         *self = SablierPlugin::new_from_config(config);
+
+        let observers = self.inner.observers.clone();
+        spawn(async move {
+            if let Err(err) = load_all_accounts(observers).await {
+                log::error!("{err}");
```
[the remark this answers is not in the document]
Comment (Adrena-Corto) [reply plugin/src/plugin.rs:58]: In whatever correct rust syntax, not sure about the proposed edit

File: plugin/src/accounts_fetcher.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use anchor_lang::Id;
+use sablier_thread_program::{
+    program::ThreadProgram,
+    state::{Thread, VersionedThread},
+};
+use solana_sdk::signature::Keypair;
+
+use crate::{error::PluginError, observers::Observers};
+
+pub async fn load_all_accounts(observers: Arc<Observers>) -> Result<(), PluginError> {
```
Comment (Adrena-Corto) [inline plugin/src/accounts_fetcher.rs:12]: Naming should be clearer, this is specifically about fetching the threadprogram Thread accounts.
And it's not even a fetch, it just create new observers?

Would name this function something like `fetch_and_observe_thread_pdas()`

Or split in two with a fetch and then later send the result outside of here to a "observe thread" (seem to make more sense but i don't know the codebase).

Also can rename the file following whatever path you go

Comment (Aursen) [reply plugin/src/accounts_fetcher.rs]: it's a small method, more work to split it currently

**Review by Adrena-Corto — APPROVED** [review]

File: plugin/src/accounts_fetcher.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    program::ThreadProgram,
+    state::{Thread, VersionedThread},
+};
+use solana_sdk::signature::Keypair;
+
+use crate::{error::PluginError, observers::Observers};
+
+pub async fn fetch_and_observe_thread_pdas(observers: Arc<Observers>) -> Result<(), PluginError> {
+    let client =
+        anchor_client::Client::new(anchor_client::Cluster::Localnet, Arc::new(Keypair::new()));
```
Comment (Adrena-Corto) [inline plugin/src/accounts_fetcher.rs:14]: Probably wrong cluster as pointed out by @adrena-orex

Comment (Adrena-Corto) [reply plugin/src/accounts_fetcher.rs:14]: Also the keypair must use the one in the local env

**Comment by Adrena-Corto** [comment]
Taking over in #70 This must be handled

**Comment by Adrena-Corto** [comment]
supplanted by #69

**Pull request closed without merging**